src/pages/Encounters/ReportBuilder/HeaderBuilder.tsx

```diff
@@ -0,0 +1,865 @@
+  );
+
+  const handleAddElement = useCallback(
+    (rowIndex: number, type: string) => {
+      onAddElement(rowIndex, type);
+    },
+    [onAddElement],
+  );
+
+  const getElementIcon = (type: string) => {
+    switch (type) {
+      case "text":
+        return "l-align-left";
+      case "image":
+        return "l-image-v";
+      case "rule":
+        return "l-minus";
+      case "datetime":
+        return "l-calender";
+      default:
+        return "l-align-left";
+    }
+  };
+
+  if (column.length === 0) {
+    return (
+      <div className="border rounded-lg p-2 bg-gray-50 flex justify-center items-center gap-2">
+        <RowButtons
+          size="lg"
+          columnIndex={columnIndex}
+          handleAddElement={handleAddElement}
+        />
+      </div>
+    );
+  }
+
+  return (
+    <div className="border rounded-lg p-2">
+      {/* Element Tabs */}
+      <div className="p-3 flex sm:flex-row flex-col justify-between gap-2">
+        {column.length > 0 && (
+          <div className="flex flex-col sm:flex-row justify-start gap-2">
+            <Select
+              value={`${activeElement}`}
+              onValueChange={(value) => toggleElement(Number(value))}
+            >
+              <SelectTrigger className="bg-green-100">
+                <SelectValue placeholder={t("REPORT_BUILDER_ADD_ELEMENT")}>
+                  <CareIcon
+                    icon={getElementIcon(column[activeElement].type)}
+                    className="w-4 h-4 text-green-900"
+                  />
+                  <span className="text-sm text-green-900">
+                    {activeElement}
+                    {". "}
+                    {t(
+                      `REPORT_BUILDER_${column[activeElement].type.toUpperCase()}`,
+                    )}
+                  </span>
+                </SelectValue>
+              </SelectTrigger>
+              <SelectContent>
+                {column.map((element, elementIndex) => (
+                  <SelectItem
+                    key={elementIndex}
+                    value={elementIndex.toString()}
+                  >
+                    {t(`REPORT_BUILDER_${element.type.toUpperCase()}`)}
+                  </SelectItem>
+                ))}
+              </SelectContent>
+            </Select>
+            <Button
+              type="button"
+              variant="secondary"
+              onClick={() => onRemoveElement(columnIndex, activeElement)}
+            >
+              <span className="text-sm">
+                {t(
+                  `REPORT_BUILDER_REMOVE_${column[activeElement].type.toUpperCase()}`,
+                )}
+              </span>
+              <Trash2Icon className="w-3 h-3" />
+            </Button>
+          </div>
+        )}
+        <Button
+          type="button"
+          size={"sm"}
+          variant="destructive"
+          onClick={() => onRemoveRow(columnIndex)}
+        >
+          <Trash2Icon className="w-3 h-3" />
```

Comment:
minor nitpick:

the tradition of doing `w-3 h-3` always manually existed only because `tailwind-merge`'s older version could not properly override classes with `size-*` when mixed with `w-* h-*`. But in todays version, it's capable of handling that as well.

```suggestion
          <Trash2Icon className="size-3" />
```